lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1339,6 +1339,12 @@ bool TypeLoweringVisitor::visitExpr(mlir::UnrealizedConversionCastOp op) {
 
 // Expand BitCastOp of aggregates
 bool TypeLoweringVisitor::visitExpr(BitCastOp op) {
+  auto isBundle = [](Type type) {
+    if (auto refType = type_dyn_cast<RefType>(type))
+      type = refType.getType();
+    return type_isa<BundleType>(type);
+  };
```

Comment:
You could use `type_isa<BundleType>(getBaseType(type))` but is it possible for RefType to be an operand of BitCast op?

Reply:
This piece of code is copied from the function `peelType` ([link](https://github.com/SpriteOvO/circt/blob/keep-aggregate-fields-order-in-cat-again/lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp#L199-L200)), Since it's already being used as the if condition ([link](https://github.com/SpriteOvO/circt/blob/7417fc55f8ca806d04f342951e3b8ae7abd586fe/lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp#L1353C52-L1353)), I guess `RefType` is possible for `op.input`?

But `op.result` doesn't use `peelType` as the condition, and I should probably replace it with just a `type_isa<BundleType>(type)`.

Reply:
`peelType` can be used for a wide range of operations but bitcast op doesn't take ref type as input. So you can drop the ref type handling.

Reply:
Makes sense, dropped it.